Lib/fontbakery/profiles/googlefonts.py

```diff
@@ -4836,6 +4837,76 @@ def com_google_fonts_check_repo_zip_files(family_directory, config):
                       f"\t* {files_list}")
 
 
+@check(
+    id = 'com.google.fonts/check/vertical_metrics',
+    conditions = ['not remote_styles', 'not is_cjk_font'],
+    rationale = """
+        This check generally enforces Google Fonts’ vertical metrics specifications.
+
+        Our documentation includes further information:
+        https://github.com/googlefonts/gf-docs/tree/main/VerticalMetrics
+    """,
+)
+def com_google_fonts_check_vertical_metrics(ttFont):
+    """Check font follows the Google Fonts vertical metric schema"""
+    from .shared_conditions import typo_metrics_enabled
+    filename = os.path.basename(ttFont.reader.file.name)
+
+    # Check necessary tables are present.
+    missing_tables = False
+    required = ["OS/2", "hhea", "head"]
+    for key in required:
+        if key not in ttFont:
+            missing_tables = True
+            yield FAIL,\
+                  Message(f'lacks-{key}',
+                          f"{filename} lacks a '{key}' table.")
+
+    if missing_tables:
+        return
+
+    font_upm = ttFont['head'].unitsPerEm
+    font_metrics = {
+        'OS/2.sTypoAscender': ttFont['OS/2'].sTypoAscender,
+        'OS/2.sTypoDescender': ttFont['OS/2'].sTypoDescender,
+        'OS/2.sTypoLineGap': ttFont['OS/2'].sTypoLineGap,
+        'hhea.ascent': ttFont['hhea'].ascent,
+        'hhea.descent': ttFont['hhea'].descent,
+        'hhea.lineGap': ttFont['hhea'].lineGap,
+        'OS/2.usWinAscent': ttFont['OS/2'].usWinAscent,
+        'OS/2.usWinDescent': ttFont['OS/2'].usWinDescent
```

Comment:
Since we've got separate checks for these and we don't test them in this check, I guess we can remove them?